dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/engine/executor/plugin/dependent/DependentTaskTracker.java

```diff
@@ -223,12 +232,40 @@ private boolean isAllDependentTaskFinished() {
-                            DEPENDENT_SPLIT,
-                            dependentKey,
-                            dependResult, dependentDate);
+                    DependentItem dependentItem = new DependentItem().fromKey(dependentKey);
+                    WorkflowDefinition workflowDefinition = processDefinitionMap.get(dependentItem.getDefinitionCode());
+                    Project project = projectCodeMap.get(workflowDefinition.getProjectCode());
+                    TaskDependentResult taskDependentResult = new TaskDependentResult();
+                    taskDependentResult.setTaskInstanceId(taskExecutionContext.getTaskInstanceId());
+                    taskDependentResult.setProjectCode(project.getCode());
+                    taskDependentResult.setWorkflowDefinitionCode(workflowDefinition.getCode());
+                    taskDependentResult.setDependentResult(dependResult);
+                    taskDependentResult.setCreateTime(new Date());
+                    taskDependentResult.setUpdateTime(new Date());
+                    if (dependentItem.getDepTaskCode() == Constants.DEPENDENT_ALL_TASK_CODE) {
+                        taskDependentResult.setTaskDefinitionCode(Constants.DEPENDENT_ALL_TASK_CODE);
+                        taskDependentResultDao.upsertTaskDependentResult(taskDependentResult);
+                        log.info(
+                                "Dependent type all task check finished, DependentResult: {}, DependentDate: {}, ProjectName: {}, WorkflowName: {}, WorkflowCode: {}, DependentCycle: {}, DependentCycleDate: {}",
+                                dependResult, dependentDate, project.getName(), workflowDefinition.getName(),
+                                workflowDefinition.getCode(), dependentItem.getCycle(), dependentItem.getDateValue());
+                    } else if (dependentItem.getDepTaskCode() == Constants.DEPENDENT_WORKFLOW_CODE) {
+                        taskDependentResult.setTaskDefinitionCode(Constants.DEPENDENT_WORKFLOW_CODE);
+                        taskDependentResultDao.upsertTaskDependentResult(taskDependentResult);
+                        log.info(
+                                "Dependent type workflow task check finished, DependentResult: {}, DependentDate: {}, ProjectName: {}, WorkflowName: {}, WorkflowCode: {}, DependentCycle: {}, DependentCycleDate: {}",
+                                dependResult, dependentDate, project.getName(), workflowDefinition.getName(),
+                                workflowDefinition.getCode(), dependentItem.getCycle(), dependentItem.getDateValue());
+                    } else {
+                        TaskDefinition taskDefinition = taskDefinitionMap.get(dependentItem.getDepTaskCode());
+                        taskDependentResult.setTaskDefinitionCode(taskDefinition.getCode());
+                        taskDependentResultDao.upsertTaskDependentResult(taskDependentResult);
```

Comment:
It would be better if we can set the dependent result into taskInstance's app_link, then we can avoid importing a table here and we can also avoid query the dependent result in other place. cc @caishunfeng

Reply:
TaskInstance's app_link is used to store yarn application ids. So I think it's not a good idea.

Reply:
It's better to add a field `runtime_context` to store all runtime context if need.

Reply:
I'll create a new table `t_ds_task_instance_context` to store `context`.